src/main/java/org/traccar/protocol/Gt06ProtocolDecoder.java

```diff
@@ -1034,6 +1035,96 @@ private Object decodeBasic(Channel channel, SocketAddress remoteAddress, ByteBuf
+
+            int mcc = buf.readUnsignedShort();
+            int msb = (mcc & 0x8000) >> 15;
+            int actualMcc = mcc & 0x7FFF;
+
+            int mnc = 0;
+            if (msb == 1) {
+                mnc = buf.readUnsignedShort();
+            } else if (msb == 0) {
+                mnc = buf.readUnsignedByte();
+            }
+
+            int cellType = buf.readUnsignedByte();
+            int cellCount = buf.readUnsignedByte();
+            Network network = new Network();
+            if (cellType == 0x00) {
+                for (int i = 0; i < cellCount; i++) {
+                    int lac = buf.readUnsignedShort();
+                    long cid = buf.readUnsignedMedium();
+                    int rssi = buf.readUnsignedByte();
+                    network.addCellTower(CellTower.from(
+                            actualMcc, mnc, lac, cid, rssi));
+                }
+            } else if (cellType == 0x01) {
+                for (int i = 0; i < cellCount; i++) {
+                    long lac = buf.readUnsignedInt();
+                    long cid = buf.readLong();
+                    int rssi = buf.readUnsignedByte();
+                    network.addCellTower(CellTower.from(
+                            actualMcc, mnc, (int) lac, cid, rssi));
+                }
+            }
+            position.set("timingAdvance", buf.readUnsignedByte());
+
+            int wifiCount = buf.readUnsignedByte();
+            for (int i = 0; i < wifiCount; i++) {
+                String macAddress = ByteBufUtil.hexDump(buf.readSlice(6)).toUpperCase();
+                int signalStrength = buf.readUnsignedByte();
+                network.addWifiAccessPoint(WifiAccessPoint.from(
+                    macAddress, signalStrength));
+            }
+
+            int event = buf.readUnsignedByte();
+            int languageDetails = buf.readUnsignedByte();
+        
+            if (languageDetails == 0x01) {
+                position.set("alertLanguage", "Chinese");
+            } else if (languageDetails == 0x02) {
+                position.set("alertLanguage", "English");
+            } else if (languageDetails == 0x00) {
+                // no reply needed
+            }
+
+            position.set(Position.KEY_EVENT, event);
+            switch (event) {
+                case 0x00 -> position.addAlarm(Position.ALARM_GENERAL);
+                case 0x01 -> position.addAlarm(Position.ALARM_SOS);
+                case 0x02 -> position.addAlarm(Position.ALARM_POWER_CUT);
+                case 0x03 -> position.addAlarm(Position.ALARM_VIBRATION);
+                case 0x04 -> position.addAlarm(Position.ALARM_GEOFENCE_ENTER);
+                case 0x05 -> position.addAlarm(Position.ALARM_GEOFENCE_EXIT);
+                case 0x06 -> position.addAlarm(Position.ALARM_OVERSPEED);
+                case 0x09 -> position.addAlarm(Position.ALARM_TOW);
+                case 0x0A -> position.addAlarm("gpsBlindSpotEnter");
```

Comment:
This is also something we don't do. You can't have random strings for alarms.

Reply:
I now have updated the existing alarms that I think fit best into those values.